File: back-end/php/task_management/api/controllers/project_controller.php
<?php

class project_controller extends controller_singletone {

    //post
    //route=addProject
    public function add_project($params) {
        return $this->service->add_project($params);
    }

    //get
    //route=getProjectById
    public function get_project_by_id($params) {
        return $this->service->get_project_by_id($params['projectId']);
    }

    //get
    //route=getAllProjects
    public function get_all_projects($params) {
        return $this->service->get_all_projects();
    }
    
    //get
    //route=getProjectByTeamLeaderId
    public function get_projects_by_team_leader_id($params) {
        return $this->service->get_projects_by_team_leader_id($params['teamLeaderId']);
    }

    //get
    //route=getProjectsReports
    public function get_projects_reports($params) {
        return $this->service->get_projects_reports();
    }

    //get
    //route=hasProjects
    public function has_projects($params) {
        return $this->service->has_projects($params['teamLeaderId']);
    }

    //post
    //checkUniqeValidations
    public function check_unique_validation($params) {
        return $this->service->check_unique_validation($params);
    }

}

File: back-end/c#/TaskManagement/API/Controllers/ProjectController.cs
using BOL;
using BLL;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Net;
using System.Net.Http;
using System.Web.Http;
using System.Web.Http.Cors;

namespace API.Controllers
{
    [EnableCors("*", "*", "*")]
    [RoutePrefix("api/project")]
    public class ProjectController : BaseController
    {
        [HttpPost]
        [Route("addProject")]
        public HttpResponseMessage AddProject(Project newProject)
        {
            try
            {
                if (ModelState.IsValid)
                {
                    bool created = ProjectService.AddProject(newProject);
                    return Request.CreateResponse(HttpStatusCode.Created, created);
                }
                List<string> errors = GetErrorList(ModelState.Values.ToList());

                return Request.CreateResponse(HttpStatusCode.BadRequest, errors);
            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message);
            }

        }

        [HttpGet]
        [Route("getProjectById")]
        public HttpResponseMessage GetProjectById(int projectId)
        {
            try
            {

                Project project = ProjectService.GetProjectById(projectId);
                ProjectService.AddDetailsToProject(project);
                return Request.CreateResponse(HttpStatusCode.OK, project);
            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.NotFound, ex.Message);
            }
        }

        [HttpGet]
        [Route("getAllProjects")]
        public HttpResponseMessage GetAllProjects()
        {
            try
            {

                List<Project> projects = ProjectService.GetAllProjects();
                return Request.CreateResponse(HttpStatusCode.OK, projects);
            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.NotFound, ex.Message);
            }
        }

        [HttpGet]
        [Route("getProjectsByTeamLeaderId")]
        public HttpResponseMessage getProjectsByTeamLeaderId(int TeamLeaderId)
        {
            try
            {
                List<Project> projects = ProjectService.GetProjectsByTeamLeaderId(TeamLeaderId);
                return Request.CreateResponse(HttpStatusCode.OK, projects);
            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.NotFound, ex.Message);
            }
        }

        [HttpGet]
        [Route("getProjectsReports")]
        public HttpResponseMessage GetProjectsReports()
        {
            try
            {

                List<Project> projects = ProjectService.GetProjectsReports();
                return Request.CreateResponse(HttpStatusCode.OK, projects);
            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.NotFound, ex.Message);
            }
        }

        [HttpGet]
        [Route("hasProjects")]
        public HttpResponseMessage HasProjects(int teamLeaderId)
        {
            try
            {
                bool hasProjects = ProjectService.HasProjects(teamLeaderId);
                return Request.CreateResponse(HttpStatusCode.OK, hasProjects);
            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.NotFound, ex.Message);
            }
        }

        [HttpPost]
        [Route("checkUniqueValidation")]
        public HttpResponseMessage CheckUniqueValidation(Project project)
        {
            try
            {
                if (ModelState.IsValid == false)
                {
                    List<string> errors = GetErrorList(ModelState.Values.ToList());
                    if (errors.Any(err => err == "ProjectName must be unique"))
                        return Request.CreateResponse(HttpStatusCode.OK, new { val = "project name must be unique" });
                }
                return Request.CreateResponse(HttpStatusCode.OK);

            }
            catch (Exception ex)
            {
                return Request.CreateErrorResponse(HttpStatusCode.InternalServerError, ex.Message);
            }
        }

    }
}
